### ISSUE TITLE
Adding custom Toolbar items no longer works with Qt5

### ISSUE THREAD

**Issue by stylekilla** [matplotlib/matplotlib#18835]
### Bug report

**Bug summary**

Every time I try to add a tool to the toolmanager, python crashes when it tries to set the icon:
```
Traceback (most recent call last):
  File "test.py", line 80, in <module>
    fig.canvas.manager.toolbar.add_tool('Show', 'navigation', 1)
  File "/Users/.../.virtualenvs/.../lib/python3.8/site-packages/matplotlib/backend_bases.py", line 3328, in add_tool
    self.add_toolitem(tool.name, group, position,
  File "/Users/.../.virtualenvs/.../lib/python3.8/site-packages/matplotlib/backends/backend_qt5.py", line 919, in add_toolitem
    button.setIcon(NavigationToolbar2QT._icon(self, image_file))
  File "/Users/.../.virtualenvs/.../lib/python3.8/site-packages/matplotlib/backends/backend_qt5.py", line 711, in _icon
    name = name.replace('.png', '_large.png')
AttributeError: 'NoneType' object has no attribute 'replace'
```

I tried to run the [example provided in the documentation](https://matplotlib.org/3.3.0/gallery/user_interfaces/toolmanager_sgskip.html) and that fails also. Although at a different point:
```
Traceback (most recent call last):
  File "test.py", line 68, in <module>
    fig.canvas.manager.toolbar.add_tool('zoom', 'foo')
AttributeError: 'NoneType' object has no attribute 'add_tool'
```

I am using:

- OSX 10.15.6 (Catalina)
- Python 3.8.6
- matplotlib==3.3.0
- PyQt5==5.15.0
- PyQt5-sip==12.8.0

Python installed via homebrew and packages installed via pip in a virtualenv.

**Code for reproduction**
The example code linked above is used, with the addition of using the Qt5Agg backend.

```
import matplotlib as mpl
mpl.use('Qt5Agg')
mpl.rcParams['toolbar'] = 'toolmanager'

import matplotlib.pyplot as plt
from matplotlib.backend_tools import ToolBase, ToolToggleBase


class ListTools(ToolBase):
	"""List all the tools controlled by the `ToolManager`."""
	# keyboard shortcut
	default_keymap = 'm'
	description = 'List Tools'

	def trigger(self, *args, **kwargs):
		print('_' * 80)
		print("{0:12} {1:45} {2}".format(
			'Name (id)', 'Tool description', 'Keymap'))
		print('-' * 80)
		tools = self.toolmanager.tools
		for name in sorted(tools):
			if not tools[name].description:
				continue
			keys = ', '.join(sorted(self.toolmanager.get_tool_keymap(name)))
			print("{0:12} {1:45} {2}".format(
				name, tools[name].description, keys))
		print('_' * 80)
		print("Active Toggle tools")
		print("{0:12} {1:45}".format("Group", "Active"))
		print('-' * 80)
		for group, active in self.toolmanager.active_toggle.items():
			print("{0:12} {1:45}".format(str(group), str(active)))


class GroupHideTool(ToolToggleBase):
	"""Show lines with a given gid."""
	default_keymap = 'G'
	description = 'Show by gid'
	default_toggled = True

	def __init__(self, *args, gid, **kwargs):
		self.gid = gid
		super().__init__(*args, **kwargs)

	def enable(self, *args):
		self.set_lines_visibility(True)

	def disable(self, *args):
		self.set_lines_visibility(False)

	def set_lines_visibility(self, state):
		for ax in self.figure.get_axes():
			for line in ax.get_lines():
				if line.get_gid() == self.gid:
					line.set_visible(state)
		self.figure.canvas.draw()


fig = plt.figure()
plt.plot([1, 2, 3], gid='mygroup')
plt.plot([2, 3, 4], gid='unknown')
plt.plot([3, 2, 1], gid='mygroup')

# Add the custom tools that we created
fig.canvas.manager.toolmanager.add_tool('List', ListTools)
fig.canvas.manager.toolmanager.add_tool('Show', GroupHideTool, gid='mygroup')


# Add an existing tool to new group `foo`.
# It can be added as many times as we want
fig.canvas.manager.toolbar.add_tool('zoom', 'foo')

# Remove the forward button
fig.canvas.manager.toolmanager.remove_tool('forward')

# To add a custom tool to the toolbar at specific location inside
# the navigation group
fig.canvas.manager.toolbar.add_tool('Show', 'navigation', 1)

plt.show()
```

**Expected outcome**
I expect the toolbar to just add my new tool without error. It used to work with:

 - Python 3.5
 - matplotlib==3.1.1
 - PyQt5==5.12.1
 - PyQt5-sip==4.19.15

Any ideas? Hopefully it's a simple fix.

**Comment by jklymak** [comment]
@fariza  any clues as to what may have changed here?  @stylekilla you may want to search on PRs and Issues with the "toolbar" label to see if anything makes sense...

**Comment by stylekilla** [comment]
I was just doing a bit more of a search... I'm not expert but it looks like the function (L:3312 in backend_bases.py) doesn't return the proper tool:
```
class ToolContainerBase: 
    ... <other funcs> ...
    def add_tool(self, tool, group, position=-1):
        """
        Add a tool to this container.

        Parameters
        ----------
        tool : tool_like
            The tool to add, see `.ToolManager.get_tool`.
        group : str
            The name of the group to add this tool to.
        position : int, default: -1
            The position within the group to place this tool.
        """
        tool = self.toolmanager.get_tool(tool)
        image = self._get_image_filename(tool.image)
        toggle = getattr(tool, 'toggled', None) is not None
        self.add_toolitem(tool.name, group, position,
                          image, tool.description, toggle)
        if toggle:
            self.toolmanager.toolmanager_connect('tool_trigger_%s' % tool.name,
                                                 self._tool_toggled_cbk)
            # If initially toggled
            if tool.toggled:
                self.toggle_toolitem(tool.name, True)
```

This leads me to think that maybe it is a tool manager problem?

**Comment by jklymak** [comment]
If you are able to setup a dev environment and run from `pip install -e .`, then if you can help debug that, that would be very helpful.  I'm just playing traffic cop here - I dont' particularly know much about the toolbar manager.

**Comment by QuLogic** [comment]
This is already fixed in 3.3.2.
